bluebinder.c

```diff
@@ -222,9 +309,29 @@ host_read_callback(
 
     proxy->host_len += len;
 
+    if (!proxy->bluetooth_hal_initialized) {
+        if (!proxy->bluetooth_on) configure_bt(proxy, TRUE);
+        fprintf(stderr, "delaying writing host command to controller until bt is up\n");
+        return true;
```

Comment:
I prefer to use `G_SOURCE_CONTINUE` and `G_SOURCE_REMOVE` instead of `TRUE`/`FALSE`. Also note that if you return `G_SOURCE_REMOVE` (`FALSE`) from a `GIOFunc` callback, you should clear watch id or something, to prevent the source from being removed twice.